core-jdk8/src/main/java/org/mapstruct/Mapping.java

```diff
@@ -180,4 +182,12 @@
      * @return Default value to set in case the source property is {@code null}.
      */
     String defaultValue() default "";
+
+    /**
+     * Decide whether we should check null or hasX method before mapping.
+     * The value on {@link Mapper} can override this one.
+     *
+     * @return strategy about how to do null or has value check
+     */
+    SourceValuePresenceCheckStrategy sourceValuePresenceCheckStrategy() default IS_NULL;
```

Comment:
is this in line with the current way of handling? I thought its is_null_in_line. I would not prefer changed default behavior.